langchain4j-core/src/test/java/dev/langchain4j/model/chat/common/ChatLanguageModelIT.java

```diff
@@ -0,0 +1,141 @@
+public abstract class ChatLanguageModelIT {
+
+    protected abstract List<ChatLanguageModel> models();
+
+    @ParameterizedTest
+    @MethodSource("models")
+    void should_answer_simple_question(ChatLanguageModel model) {
+
+        // given
+        model = spy(model);
+
+        ChatRequest chatRequest = ChatRequest.builder()
+                .messages(UserMessage.from("What is the capital of Germany?"))
+                .build();
+
+        // when
+        ChatResponse chatResponse = model.chat(chatRequest);
+
+        // then
+        AiMessage aiMessage = chatResponse.aiMessage();
+        assertThat(aiMessage.text()).containsIgnoringCase("Berlin");
+        assertThat(aiMessage.toolExecutionRequests()).isNull();
+
+        TokenUsage tokenUsage = chatResponse.tokenUsage();
+        assertThat(tokenUsage.inputTokenCount()).isGreaterThan(0);
+        assertThat(tokenUsage.outputTokenCount()).isGreaterThan(0);
+        assertThat(tokenUsage.totalTokenCount())
+                .isEqualTo(tokenUsage.inputTokenCount() + tokenUsage.outputTokenCount());
+
+        if (assertFinishReason()) {
+            assertThat(chatResponse.finishReason()).isEqualTo(STOP);
+        }
+    }
+
+    @EnabledIf("supportsTools")
+    @ParameterizedTest
+    @MethodSource("models")
+    void should_call_a_tool(ChatLanguageModel model) {
+
+        // given
+        model = spy(model);
+
+        ToolSpecification toolSpecification = ToolSpecification.builder()
+                .name("weather")
+                .parameters(JsonObjectSchema.builder()
+                        .addStringProperty("city")
+                        .build())
+                .build();
+
+        ChatRequest chatRequest = ChatRequest.builder()
+                .messages(UserMessage.from("What is the weather in Munich?"))
+                .toolSpecifications(toolSpecification)
+                .build();
+
+        // when
+        ChatResponse chatResponse = model.chat(chatRequest);
+
+        // then
+        AiMessage aiMessage = chatResponse.aiMessage();
+        assertThat(aiMessage.toolExecutionRequests()).hasSize(1);
+
+        ToolExecutionRequest toolExecutionRequest = aiMessage.toolExecutionRequests().get(0);
+        assertThat(toolExecutionRequest.name()).isEqualTo(toolSpecification.name());
+        assertThat(toolExecutionRequest.arguments()).isEqualToIgnoringWhitespace("{\"city\":\"Munich\"}");
+
+        TokenUsage tokenUsage = chatResponse.tokenUsage();
+        assertThat(tokenUsage.inputTokenCount()).isGreaterThan(0);
+        assertThat(tokenUsage.outputTokenCount()).isGreaterThan(0);
+        assertThat(tokenUsage.totalTokenCount())
+                .isEqualTo(tokenUsage.inputTokenCount() + tokenUsage.outputTokenCount());
+
+        if (assertFinishReason()) {
+            assertThat(chatResponse.finishReason()).isEqualTo(TOOL_EXECUTION);
+        }
+    }
+
+    protected boolean supportsTools() {
+        return true; // TODO check model capability instead?
```

Comment:
Once `ToolMode` will be added to the API, we will definitely test different modes in these tests.

This comment is for future me to change the logic to check for `Capability.TOOLS` (once it will be added)